host/xtest/xtest_20000.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2014, STMicroelectronics International N.V.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License Version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <adbg.h>
+#include <xtest_test.h>
+#include <xtest_helpers.h>
+
+#include <tee_client_api.h>
+#include <ta_storage.h>
+#include <tee_api_defines.h>
+#include <tee_api_types.h>
+
+/*
+ * defines based on definition of
+ * unsecure GP file header (struct tee_svc_storage_head)
+ */
+#define HEADER_OFFSET            0
+#define HEADER_SIZE              (sizeof(uint32_t)*4)
+#define HEADER_HEADSIZE_OFFSET   (HEADER_OFFSET + sizeof(uint32_t)*1)
+#define HEADER_DATASIZE_OFFSET   (HEADER_OFFSET + sizeof(uint32_t)*3)
+
+#define META_OFFSET              HEADER_SIZE
+#define META_SIZE                0
+#define PAYLOAD_OFFSET           (META_OFFSET + META_SIZE)
+
+/*
+ * defines based on definition of
+ * secure Linaro enc_fs file header (struct enc_fs_file_header)
+ */
+#define SEC_HEADER_OFFSET        PAYLOAD_OFFSET
+#define SEC_HEADER_SIZE          (16+12+16)
+#define SEC_HEADER_KEY_OFFSET    SEC_HEADER_OFFSET
+#define SEC_HEADER_IV_OFFSET     (SEC_HEADER_OFFSET + 16)
+#define SEC_HEADER_TAG_OFFSET    (SEC_HEADER_OFFSET + 16+12)
+
+#define ENCDATA_OFFSET           (PAYLOAD_OFFSET + SEC_HEADER_SIZE)
+
+#define CORRUPT_HEADSIZE_OFFSET  HEADER_HEADSIZE_OFFSET
+#define CORRUPT_DATASIZE_OFFSET  HEADER_DATASIZE_OFFSET
+#define CORRUPT_KEY_OFFSET       SEC_HEADER_KEY_OFFSET
+#define CORRUPT_IV_OFFSET        SEC_HEADER_IV_OFFSET
+#define CORRUPT_TAG_OFFSET       SEC_HEADER_TAG_OFFSET
+#define CORRUPT_DATA_OFFSET      (ENCDATA_OFFSET + MIN(sizeof(data_01)/2, 10))
+
+#ifndef MIN
+#define MIN(a,b) ((a)<(b) ? (a) : (b))
+#endif
+
+#define SWAP_BYTES_16(w16) ((((w16) & 0xFF00) >> 8) | (((w16) & 0xFF) << 8))
+#define SWAP_BYTES_32(w32) ((((w32) & 0xFF000000) >> 24) |\
+			    (((w32) & 0xFF0000) >> 8) |\
+			    (((w32) & 0xFF00) << 8) |\
+			    (((w32) & 0xFF) << 24))
+
+uint8_t file_01[] = {
+'f', 'i', 'l', 'e', '_', '0', '1'
+};
```

Comment:
Would be more readable as:
```
uint8_t file_01[7] = "file_01";
```
(or `file_01[] =` if you don't mind a `\0` being added)